src/btree/bt_sync.c

```diff
@@ -152,6 +152,8 @@ __sync_file(WT_SESSION_IMPL *session, int syncop)
 				leaf_bytes += page->memory_footprint;
 				++leaf_pages;
 			}
+			if (txn->isolation == WT_ISO_READ_COMMITTED)
+				__wt_txn_get_snapshot(session);
```

Comment:
We need this to make the "read-committed" checkpoint of the metadata work.  Otherwise there are paths into `__wt_meta_track_off` without a snapshot that try to do visibility checks.